src/Database/DBA.php

```diff
@@ -543,33 +543,38 @@ public static function buildCondition(array &$condition = [])
 						$condition_string .= " AND ";
 					}
 					if (is_array($value)) {
-						/* Workaround for MySQL Bug #64791.
-						 * Never mix data types inside any IN() condition.
-						 * In case of mixed types, cast all as string.
-						 * Logic needs to be consistent with DBA::p() data types.
-						 */
-						$is_int = false;
-						$is_alpha = false;
-						foreach ($value as $single_value) {
-							if (is_int($single_value)) {
-								$is_int = true;
-							} else {
-								$is_alpha = true;
+						if (count($value)) {
+							/* Workaround for MySQL Bug #64791.
+							 * Never mix data types inside any IN() condition.
+							 * In case of mixed types, cast all as string.
+							 * Logic needs to be consistent with DBA::p() data types.
+							 */
+							$is_int = false;
+							$is_alpha = false;
+							foreach ($value as $single_value) {
+								if (is_int($single_value)) {
+									$is_int = true;
+								} else {
+									$is_alpha = true;
+								}
 							}
-						}
 
-						if ($is_int && $is_alpha) {
-							foreach ($value as &$ref) {
-								if (is_int($ref)) {
-									$ref = (string)$ref;
+							if ($is_int && $is_alpha) {
+								foreach ($value as &$ref) {
+									if (is_int($ref)) {
+										$ref = (string)$ref;
+									}
 								}
+								unset($ref); //Prevent accidental re-use.
 							}
-							unset($ref); //Prevent accidental re-use.
-						}
 
-						$new_values = array_merge($new_values, array_values($value));
-						$placeholders = substr(str_repeat("?, ", count($value)), 0, -2);
-						$condition_string .= self::quoteIdentifier($field) . " IN (" . $placeholders . ")";
+							$new_values = array_merge($new_values, array_values($value));
+							$placeholders = substr(str_repeat("?, ", count($value)), 0, -2);
+							$condition_string .= self::quoteIdentifier($field) . " IN (" . $placeholders . ")";
+						} else {
+							// Empty value array isn't supported by IN and is logically equivalent to no match
+							$condition_string .= "FALSE";
```

Comment:
Please add a space before the "FALSE". Otherwise this will most likely cause an SQL error.

Reply:
Nope, the space is added on line 543. If you look at the other `$condition_string` concatenation cases, none have a leading space either.

Reply:
Ah, I missed that.